### PR TITLE
SEC-2258 - Added new method update TTL attribute to table manager

### PR DESCRIPTION
**Pull request by JohnOxleySage**
This also swapped the tests to using Localstack over a local docker instance as local docker doesn't support the `update_ttl_attribute` method.

### CODE REVIEW

File: spec/dynamodb_framework/dynamodb_table_manager_spec.rb [hunk 1]
```diff
@@ -117,6 +117,29 @@
...
+    end
+
+
+    builder = DynamoDbFramework::AttributesBuilder.new
+    builder.add(:id, :S)
+    builder.add(:name, :S)
+    builder.add(:number, :N)
+    builder.add(:ttl_date, :N)
+
+    subject.create('update_ttl_test', attributes_builder.attributes, :id)
```
Comment (adamgeorgeson) [inline spec/dynamodb_framework/dynamodb_table_manager_spec.rb:135]: I'm not sure what this test is testing?
 - `builder` is being setup up lines 129..133 but then not used
 - No checks that the ttl attribute has been updated from false to true that I can see.

File: lib/dynamodb_framework/dynamodb_table_manager.rb [hunk 2]
```diff
@@ -116,6 +116,25 @@ def update_index_throughput(table_name, index_name, read_capacity, write_capacit
       DynamoDbFramework.logger.info "[#{self.class}] -Table: [#{table_name}] updated."
     end
 
+    def update_ttl_attribute(table_name, time_to_live_status, attribute_name)
```
Comment (adamgeorgeson) [inline lib/dynamodb_framework/dynamodb_table_manager.rb:119]: `enabled` rather than `time_to_live_status`? wdyt?

Comment (JohnOxleySage) [reply lib/dynamodb_framework/dynamodb_table_manager.rb:119]: Perfectly sensible, `time_to_live_status` was a leftover from when I was trying to use a different dynamodb client method I'll update it.

File: README.md [hunk 3]
```diff
@@ -555,6 +555,36 @@ This method is called to update the throughput required by an index.
...
     manager.update_index_throughput('event_tracking', 'type_index', 50, 20)
 
+### #update_ttl_attribute
+
+This method is called to update the ttl attribute of a table.
+
+**Params** 
+
+ - **table_name** [String] [Required] This is the name of the table the attribute belongs to.
+ - **time_to_live_status** [Boolean] [Required] This is true to turn TTL on on the table or false to turn it off.
```
Comment (adamgeorgeson) [inline README.md:565]:
```suggestion
 - **enabled** [Boolean] [Required] This is true to turn TTL on the table or false to turn it off.
```

File: lib/dynamodb_framework/dynamodb_table_manager.rb [hunk 4]
```diff
@@ -261,6 +263,24 @@ def wait_until_index_dropped(table_name, index_name)
...
+        status = get_ttl_status(table_name)
+
+        if status == 'ENABLED' || 'DISABLED'
+          return
+        end
+
+        sleep(5)
+      end
+
+      raise "Timeout occurred while waiting for table: #{table_name}, to become active."
```
Comment (adamgeorgeson) [inline lib/dynamodb_framework/dynamodb_table_manager.rb:280]:
```suggestion
      raise "Timeout occurred while waiting for table: #{table_name}, to update TTL status."
```

**Pull request merged**